### PR TITLE
feat: Add `Hide mock location` patch

### PR DESCRIPTION
**Pull request by epireyn**

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import app.revanced.patches.all.misc.transformation.IMethodCall
+import app.revanced.patches.all.misc.transformation.Instruction35cInfo
+import app.revanced.patches.all.misc.transformation.filterMapInstruction35c
+import com.android.tools.smali.dexlib2.iface.ClassDef
+import com.android.tools.smali.dexlib2.iface.Method
+import com.android.tools.smali.dexlib2.iface.instruction.Instruction
+
+@Patch(
+    name = "Hide Mock Location",
+    description = "Prevent app from knowing whether the location is mocked",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt:17]:
```suggestion
    description = "Prevent app from knowing whether the location is mocked.",
```

File: src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    requiresIntegrations = true,
+    use = true
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt:19]:
```suggestion
    use = false
```

File: src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import app.revanced.patches.all.misc.transformation.BaseTransformInstructionsPatch
+import app.revanced.patches.all.misc.transformation.IMethodCall
+import app.revanced.patches.all.misc.transformation.Instruction35cInfo
+import app.revanced.patches.all.misc.transformation.filterMapInstruction35c
+import com.android.tools.smali.dexlib2.iface.ClassDef
+import com.android.tools.smali.dexlib2.iface.Method
+import com.android.tools.smali.dexlib2.iface.instruction.Instruction
+
+@Patch(
+    name = "Hide Mock Location",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt:16]:
```suggestion
    name = "Hide mock location",
```

File: src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt (hunk 1 above)
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt:17]:
```suggestion
    description = "Spoof the app when mocking the location.",
```

@LisoUseInAIKyrios maybe you have a better description.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by epireyn** [comment]
I'm currently removing the dependency on integrations.

File: src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    ): Instruction35cInfo? {
+        if (instruction.opcode != Opcode.INVOKE_VIRTUAL) {
+            return null
+        }
+
+        val invokeInstruction = instruction as Instruction35c
+        val methodRef = invokeInstruction.reference as MethodReference
+        val methodCall = fromMethodReference<MethodCall>(methodRef) ?: return null
+
+        return Instruction35cInfo(methodCall, invokeInstruction, instructionIndex)
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/location/hide/HideMockLocationPatch.kt:41]: A Pair of invokeInstruction and instructionIndex would suffice here: `BaseTransformInstructionsPatch<Pair<Instruction, Index>()`

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Something here is not right. You don't check for the correct call in filterMap. Please review the patch

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**